### ISSUE TITLE
Tree not cleanly installed for torrent-stream

### ISSUE THREAD

**Issue by TheAlphaNerd** [npm/npm#11062]
osx 10.10
node v5.4.0
npm v3.3.12

to reproduce

`git clone https://github.com/mafintosh/torrent-stream.git`
`npm install`
`npm ls`

```
npm ERR! missing: hat@0.0.3, required by torrent-stream@1.0.2
npm ERR! missing: bencode@^0.7.0, required by parse-torrent-file@2.1.4
npm ERR! missing: bittorrent-tracker@^7.0.0, required by torrent-discovery@5.2.0
npm ERR! missing: bencode@^0.7.0, required by bittorrent-dht@6.2.0
npm ERR! missing: bencode@^0.7.0, required by k-rpc-socket@1.4.3
```

This works as expected on LTS (npm v2.x). What is even weirder is that if you checkout v1.0.1 of torrent-stream everything gets resolved. [This commit](https://github.com/mafintosh/torrent-stream/commit/b990ef1ef9c5655b32921253dd8604bd2733bba0) pruning deps broke everything. Manually install hat `npm i hat` makes `npm ls` resolve error free.

/cc @mafintosh

**Comment by iarna** [comment]
I tried this w/ `npm@3.5.4`, and `https://github.com/mafintosh/torrent-stream.git#61b7bd9` from and "only" `bittorrent-tracker` fails to install:

```
npm ERR! missing: bittorrent-tracker@^7.0.0, required by torrent-discovery@5.3.0
```

Incidentally, a second run of `npm install` corrects the tree:

**Comment by iarna** [comment]
For fun, I tried `npm install https://github.com/mafintosh/torrent-stream.git#61b7bd9` and bafflingly that worked just fine.

**Comment by TheAlphaNerd** [comment]
we are deep into the weirdness

**Comment by benoitvallon** [comment]
I mentioned missing packages on first `npm install` as well in this issue https://github.com/npm/npm/issues/10985 it seems to me that those bugs have similarities.

`git clone https://github.com/benoitvallon/react-native-nw-react-calculator.git`
`cd react-native-nw-react-calculator`
`npm install`
`npm ls`

Here is what is missing after the first `npm install`
```
npm ERR! missing: shebang-regex@^1.0.0, required by babel-core@6.4.0
npm ERR! missing: slash@^1.0.0, required by babel-core@6.4.0
npm ERR! missing: babel@^5.4.7, required by fbjs-scripts@0.4.0
```
Everything is here after the second run.

In my case, the bug is reproductible and I get the same packages missing every times.

**Comment by lxe** [comment]
Interesting, 

```
node_modules/torrent-discovery/package.json:    "bittorrent-tracker": "^7.0.0",
```

While the root package wants this as a **devDependency**:
```
"bittorrent-tracker": "^2.6.0",
```

I think this is the same bug as #10119 #10381, etc where dev

**Comment by schmod** [comment]
Very likely also a dupe of #10727

**Comment by zkat** [comment]
This is quite probably fixed as of npm@4.0.5 thanks to #12811. Cheers!
